File: packages/aws-cdk/lib/init-templates/v2/app/javascript/test/%name%.test.template.js
const cdk = require('aws-cdk-lib');
const { Template } = require('@aws-cdk/assertions-alpha');
const %name.PascalCased% = require('../lib/%name%-stack');

test('Empty Stack', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new %name.PascalCased%.%name.PascalCased%Stack(app, 'MyTestStack');
  // THEN
  const template = Template.fromStack(stack);
  template.templateMatches({});
});

File: packages/aws-cdk/lib/init-templates/v2/app/go/%name%_test.template.go
package main

import (
	"testing"

	"github.com/aws/aws-cdk-go/awscdk/v2"
	assertions "github.com/aws/aws-cdk-go/awscdkassertionsalpha/v2"
	"github.com/aws/jsii-runtime-go"
)

func Test%name.PascalCased%Stack(t *testing.T) {
	// GIVEN
	app := awscdk.NewApp(nil)

	// WHEN
	stack := New%name.PascalCased%Stack(app, "MyStack", nil)

	// THEN
	template := assertions.Template_FromStack(stack)

	template.HasResourceProperties(jsii.String("AWS::SNS::Topic"), map[string]interface{}{
		"DisplayName": "MyCoolTopic",
	})
}
